### PR TITLE
MD5 Checksum changes

### PR DESCRIPTION
**Pull request by shdeshpa07**
## Jira 

* [OADP-4277](https://issues.redhat.com/browse/OADP-4277)

Added documentation for MD5 checksum algorithms 

##  Version

* OCP 4.13 → OCP 4.17

## Preview

*  [AWS MD5 checksum algorithms](https://81329--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.html#oadp-configuring-aws-md5sum_installing-oadp-aws)

## QE Review

* [ ] QE has approved this change.

### CODE REVIEW

**Comment by shdeshpa07** [comment]
@anarnold97 - Can you please review my PR? Thanks.

File: modules/oadp-configuring-aws-md5sum.adoc [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      defaultPlugins:
+      - openshift
+      - aws
+      - csi
+----
+<1> Specify the `checksumAlgorithm`. In this example, the `checksumAlgorithm` field is set to an empty value. You can select an option from the following list: `CRC32`, `CRC32C`, `SHA1`, `SHA256`.
+
+[IMPORTANT]
+====
+If you are using Noobaa as the object storage provider and you do not set the `spec.backupLocation.velero.config.checksumAlgorithm` field in the DPA, an empty value of `checksumAlgorithm` is added to the BSL configuration.
```
Comment (anarnold97) [inline modules/oadp-configuring-aws-md5sum.adoc:71]:
```suggestion
If you are using Noobaa as the object storage provider, and you do not set the `spec.backupLocation.velero.config.checksumAlgorithm` field in the DPA, an empty value of `checksumAlgorithm` is added to the BSL configuration. 
```

**Comment by shdeshpa07** [comment]
@shubham-pampattiwar  @stillalearner  -  Could you please review this PR? Thanks.

File: modules/oadp-configuring-aws-md5sum.adoc [hunk 2]
```diff
@@ -0,0 +1,75 @@
+// Module included in the following assemblies:
+//
+// * backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.adoc
+
+
+:_mod-docs-content-type: PROCEDURE
+[id="oadp-configuring-aws-md5sum_{context}"]
+= Configuring the backup storage location with MD5 checksum algorithm
+
+You can configure the Backup Storage Location (BSL) in the Data Protection Application (DPA) to use a MD5 checksum algorithm for {aws-short} S3 and S3 compatible storage providers. The checksum algorithm calculates the checksum for uploading and downloading objects to S3. You can use one of the following options to set the `checksumAlgorithm` field in the `spec.backupLocation.velero.config.checksumAlgorithm` section of the DPA.
```
Comment (stillalearner) [inline modules/oadp-configuring-aws-md5sum.adoc:10]:
```suggestion
You can configure the Backup Storage Location (BSL) in the Data Protection Application (DPA) to use a MD5 checksum algorithm for {aws-short} S3 and S3 compatible storage providers. The checksum algorithm calculates the checksum for uploading and downloading objects to S3. You can use one of the following options to set the `checksumAlgorithm` field in the `spec.backupLocations.velero.config.checksumAlgorithm` section of the DPA.
```

Comment (shubham-pampattiwar) [reply modules/oadp-configuring-aws-md5sum.adoc:10]: +1

File: modules/oadp-configuring-aws-md5sum.adoc [hunk 3]
```diff
@@ -0,0 +1,75 @@
+// Module included in the following assemblies:
+//
+// * backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.adoc
+
+
+:_mod-docs-content-type: PROCEDURE
+[id="oadp-configuring-aws-md5sum_{context}"]
+= Configuring the backup storage location with MD5 checksum algorithm
+
+You can configure the Backup Storage Location (BSL) in the Data Protection Application (DPA) to use a MD5 checksum algorithm for {aws-short} S3 and S3 compatible storage providers. The checksum algorithm calculates the checksum for uploading and downloading objects to S3. You can use one of the following options to set the `checksumAlgorithm` field in the `spec.backupLocations.velero.config.checksumAlgorithm` section of the DPA.
```
Comment (ShaunaDiaz) [inline modules/oadp-configuring-aws-md5sum.adoc:10]:
```suggestion
You can configure the Backup Storage Location (BSL) in the Data Protection Application (DPA) to use a MD5 checksum algorithm for both Amazon Simple Storage Service (Amazon S3) and S3-compatible storage providers. The checksum algorithm calculates the checksum for uploading and downloading objects to Amazon S3. You can use one of the following options to set the `checksumAlgorithm` field in the `spec.backupLocations.velero.config.checksumAlgorithm` section of the DPA.
```
Hm, it appears that Amazon calls this "Amazon Simple Storage Service (Amazon S3)", see https://aws.amazon.com/s3/

File: modules/oadp-configuring-aws-md5sum.adoc [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+====
+You can also set the `checksumAlgorithm` field to an empty value to skip the MD5 checksum check. 
+
+If you do not set a value for the `checksumAlgorithm` field, then the default value is set to `CRC32`.
+====
+
+.Prerequisites
+
+* You have installed the {oadp-short} Operator.
+* You have configured {aws-short} S3, or S3 compatible object storage as a backup location.
```
Comment (ShaunaDiaz) [inline modules/oadp-configuring-aws-md5sum.adoc:27]: Here as well ensure that we're referring to the correct thing.

**Review by ShaunaDiaz — COMMENTED** [review]
Looks good, few minor comments.

**Comment by ShaunaDiaz** [comment]
@shdeshpa07 You'll need an engineering SME and QE sign-off to merge this.

**Comment by shdeshpa07** [comment]
@shubham-pampattiwar  and @stillalearner - Please approve the changes. Cannot proceed without the approval. THanks.

**Review by stillalearner — APPROVED** [review]
lgtm

**Review by shubham-pampattiwar — APPROVED** [review]

File: modules/oadp-configuring-aws-md5sum.adoc (hunk 4 above)
Comment (shdeshpa07) [reply modules/oadp-configuring-aws-md5sum.adoc:27]: Done. thanks.

**Comment by shdeshpa07** [comment]
> @shdeshpa07 You'll need an engineering SME and QE sign-off to merge this.

Thank you Shauna. Dev and QE have approved the PR.

**Pull request merged**